handlers/istio_config.go

```diff
@@ -73,11 +73,25 @@ func IstioConfigList(w http.ResponseWriter, r *http.Request) {
 
 	if includeValidations {
 		// We don't filter by service and workload when calling validations, because certain validations require fetching all types to get the correct errors
-		// when namespace is empty, validaions should be done per all namespaces to apply object filters
-		istioConfig.IstioValidations, err = business.Validations.GetValidations(r.Context(), cluster, namespace, "", "")
-		if err != nil {
-			handleErrorResponse(w, err)
-			return
+		if namespace == "" {
+			// when namespace is empty, validations should be done per namespaces to apply exportTo configs
+			loadedNamespaces, _ := business.Namespace.GetClusterNamespaces(r.Context(), cluster)
+			istioConfig.IstioValidations = models.IstioValidations{}
+			for _, ns := range loadedNamespaces {
+				nsValidations, nsErr := business.Validations.GetValidations(r.Context(), cluster, ns.Name, "", "")
+				if nsErr != nil {
+					handleErrorResponse(w, nsErr)
+					return
+				}
+				istioConfig.IstioValidations.MergeValidations(nsValidations)
+			}
```

Comment:
Why is the being done in the handler and not in the validations? It seems like when namespace is empty, `business.Validations.GetValidations` should validate `exportTo` across namespaces without having to loop through every namespace. Otherwise calling `GetValidations` with an empty namespace is never valid?

Reply:
Yes, calling `GetValidations` with empty `namespace` is wrong, even though it does not throw an exception, it will not consider `exportTo` of configs and will loop through all configs of all namespaces and do bulk list validation.
When having `namespace` value set, the validation will be done for configs which are `exportedTo` provided `naemspace`, and results will be correct.

Reply:
> it will not consider exportTo of configs and will loop through all configs of all namespaces and do bulk list validation.

So if you pass `""` for namespace then `exportTo` will be ignored? This seems like an issue inside `GetValidations` then. Can we fix the problem there? Other places that call `GetValidations` outside of this handler may suffer from the same issue. Doesn't the `services` page have an `IncludeValidations` option?

Reply:
Yes, when you pass `""` for namespace then `exportTo` will be ignored, which is an issue.
Only other place where `GetValidations` are called is `ConfigValidationSummary` for Overview page. And here is inconsistency: Overview page calls validations then count Istio config objects by the validation results, so the `obj1` which is in `ns1` but exported to `ns2` might cause inconsistency: When namespace is empty in GetValidations call, than it will ignore `exportTo` and will show validations for `obj1` in`ns1` Overview card. When for Overview page we pass namespace param in "GetValidations", then `obj1` validations will be ignored in `ns1` Overview card, but then you go to 'ns1' Istio config list, then "obj1" is there with validations. Because Istio config list lists objects then adds validations to them.
Services page does not have config validations, but a service validations.
For the rest of the calls namespace param has a value.